### DISCUSSION TITLE
Guidance on how to configure milvus in an existing EKS with pod identity

### DISCUSSION THREAD

**Discussion by Yc-Chen** [milvus-io/milvus#35780] [Q&A and General discussion] [answered]
Hi! I have been looking at milvus deployment guide but it needs you to start with creating an eks cluster. What if I already have an EKS cluster? And how to make it authenticate to S3 using [pod identity](https://aws.amazon.com/blogs/containers/amazon-eks-pod-identity-a-new-way-for-applications-on-eks-to-obtain-iam-credentials/)?

I created a simple gist here to show how kubernetes deployment yaml could look like using pod identity: https://gist.github.com/Yc-Chen/0171e4407daad75552a7077316e3932c

Once you configure `serviceAccountName: s3-access-account`, the pod will automatically receive environment variables: `AWS_CONTAINER_CREDENTIALS_FULL_URI`, `AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE`. You can see the full list [here](https://aws.amazon.com/blogs/containers/amazon-eks-pod-identity-a-new-way-for-applications-on-eks-to-obtain-iam-credentials/).

Will milvus automatically read from these environment variables to authenticate to aws?

**Comment by yhmo** [comment]
Wait for @LoveEachDay comment.

Did you read this doc? https://milvus.io/docs/eks.md#Deploy-a-Milvus-Cluster-on-EKS

**Reply by LoveEachDay** [reply, marked as answer]
@Yc-Chen Yes, milvus pod will honor the automatically rejected environment variables. You'd add the following configurations to milvus.yaml :
```
minio:
  cloudProvider: aws
  useIAM: true
  useSSL: true
```

**Reply by Yc-Chen** [reply]
Thanks, although I could not verify as I failed with some other errors. From many pods I got exec format error like [here](https://stackoverflow.com/questions/77287655/how-to-solve-docker-container-exec-format-error-for-running-entrypoint). If I cannot solve it, I will create another discussion!
